core-tests/src/test/scala/au/com/agiledigital/rest/security/BodyParsersSpec.scala

```diff
@@ -33,20 +40,105 @@ class BodyParsersSpec(implicit ev: ExecutionEnv) extends BaseSpec with DataTable
       )
     )
 
+    val expectedUnsafeObject = Json.obj(
+      "foo" -> Json.obj(
+        "bar" -> "<span>bar</span> baz",
+        "barUnsafe" -> "<img src=\"foo\"><span>bar</span><script>alert('baz');</script> baz"
+      ),
+      "fooUnsafe" -> Json.obj(
+        "bar" -> "<img src=\"foo\"><span>bar</span><script>alert('baz');</script> baz"
+      )
+    )
+
+    // @formatter:off
     // format: OFF
-    "clean JSON values using the HTML whitelist" ||
-      "description"                       || "json"               || "expected"          |>
-      "leave non-HTML strings alone"      !! JsString("bar baz")  !! JsString("bar baz") |
-      "remove HTML elements"              !! jsString             !! JsString("bar baz") |
-      "apply recursively to JSON arrays"  !! jsArray              !! expectedArray       |
-      "apply recursively to JSON objects" !! jsObject             !! expectedObject      |
-      "leave JSON numbers alone"          !! JsNumber(42)         !! JsNumber(42)        |
-      "leave JSON null alone"             !! JsNull               !! JsNull              |
-      "leave JSON bools alone"            !! JsBoolean(false)     !! JsBoolean(false)    |> {
-      (description, json, expected) => {
+    "clean JSON values using the specified HTML whitelist" ||
+      "description"                       || "json"               || "expected clean all" || "expected clean unsafe"          |>
+      "leave non-HTML strings alone"      !! JsString("bar baz")  !! JsString("bar baz")  !! JsString("bar baz")              |
+      "remove HTML elements"              !! jsString             !! JsString("bar baz")  !! JsString("<span>bar</span> baz") |
+      "apply recursively to JSON arrays"  !! jsArray              !! expectedArray        !! expectedUnsafeArray              |
+      "apply recursively to JSON objects" !! jsObject             !! expectedObject       !! expectedUnsafeObject             |
+      "leave JSON numbers alone"          !! JsNumber(42)         !! JsNumber(42)         !! JsNumber(42)                     |
+      "leave JSON null alone"             !! JsNull               !! JsNull               !! JsNull                           |
+      "leave JSON bools alone"            !! JsBoolean(false)     !! JsBoolean(false)     !! JsBoolean(false)                 |> {
+      (description, json, expectedCleanAll, expectedUnsafe) => {
+            // @formatter:on
             // format: ON
-            BodyParsers.clean(json) must beEqualTo(expected)
+            BodyParsers.clean(json) must beEqualTo(expectedCleanAll)
+
+            BodyParsers.cleanUnsafe(json) must beEqualTo(expectedUnsafe)
           }
       }
   }
+
+  "BodyParsers whitelistingJson" should {
```

Comment:
Please make this something that would make sense to a human reading it out loud. `"Parsing JSON to a value after applying a white listing filter" should`.